### PR TITLE
fix: buildx in k8s pod with kube client config in cluster

### PR DESCRIPTION
**Pull request by morlay**
Signed-off-by: Wang Jinglei <morlay.null@gmail.com>

use kube config in cluster not work with last PR https://github.com/docker/buildx/pull/368

sorry for checking so late.

### CODE REVIEW

File: commands/util.go [hunk 1]
```diff
@@ -203,9 +203,23 @@ func driversForNodeGroup(ctx context.Context, dockerCli command.Cli, ng *store.N
 						logrus.Error(err)
 					}
 				}
+
+				tryToUseKubeConfigInCluster := false
 				if kcc == nil {
-					kcc = driver.KubeClientConfigInCluster{}
+					tryToUseKubeConfigInCluster = true
+				} else {
+					if _, err := kcc.ClientConfig(); err != nil {
```
Comment (AkihiroSuda) [inline commands/util.go:211]: Wouldn't hurt to logrus.Debug(err) here?

**Review by AkihiroSuda — APPROVED** [review]

File: commands/util.go (hunk 1 above)
Comment (morlay) [reply commands/util.go:211]: i found kcc not be nil in cluster pod.

but not found other way to check local kube config exists.

not sure which logrus.Debug(err) you means? I haven't notice logrus.Debug in Client Config

Comment (AkihiroSuda) [reply commands/util.go:211]: I thought printing the err with `logrus.Debug(err)` might be useful for debugging potential failures, but either is fine to me

**Comment by tonistiigi** [comment]
@morlay needs rebase (and maybe update)

**Comment by morlay** [comment]
@tonistiigi updated

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**